SwinMM/Initialize the SwinMM project

Zi-hao-Wei opened the pull request:
Hi, I add the SwinMM project into this repository.

wyli commented on SwinMM/WORD/dataset/dataset12_WORD_0.1.json:
thanks for the PR! could you please zip the json files and put them in a google drive folder? then you can provide instructions for the users to download them like this example https://github.com/Project-MONAI/research-contributions/tree/main/SwinUNETR/BRATS21#models

wyli commented on SwinMM/WORD/README.md:
```diff
@@ -0,0 +1,42 @@
+
+## Training
+
+### FIXME(outdated)
```
I guess this would need some updates

Zi-hao-Wei replied:
This file is out of date and I have removed it. All the training details can be find in the TRAINING.md

wyli reviewed (approved):
thanks for the PR, it looks great, please help fix those two minor issues

Zi-hao-Wei commented:
Hi, I have fix the problems. Could you please check it again? Thanks.

tangy5 commented:
This is great, thank you.

tangy5 commented:
This PR looks good to me:
HI @Zi-hao-Wei can you double check:
1. Whether there are duplicated or unused lines of code in your scripts.
2. Whethere there are additional license. 
3. I saw the WORD_finetuned.pt is actually the BTCV finetined checkpoint download link, is that correct? 
4. Can you add MONAI paper reference to the README.md? 
```
@article{cardoso2022monai,
  title={Monai: An open-source framework for deep learning in healthcare},
  author={Cardoso, M Jorge and Li, Wenqi and Brown, Richard and Ma, Nic and Kerfoot, Eric and Wang, Yiheng and Murrey, Benjamin and Myronenko, Andriy and Zhao, Can and Yang, Dong and others},
  journal={arXiv preprint arXiv:2211.02701},
  year={2022}
}
```

Thanks

wyli commented:
/black

Zi-hao-Wei commented:
Thanks for your reply. To address your concerns:
1. No duplicated or unused lines in the scripts.
2. No additional license.
3. This is a naming error and I have fixed the sharing link. Thank you for pointing out.
4. I have added it to the Citation part in README.md

wyli commented on SwinMM/Pretrain/utils/dataset_in_memory.py:
```diff
@@ -0,0 +1 @@
+../../WORD/utils/dataset_in_memory.py
```
this doesn't look like a python script file though, could you please check the filename?

wyli replied:
I commented out this line and the line in the file https://github.com/Zi-hao-Wei/research-contributions/blob/5e51052958d8309a9c153d480445f7bb37b1f78c/SwinMM/Pretrain/utils/view_transforms.py please feel free to create new PRs to modify them if I misunderstood... thanks!
